server/app/auth_test.go

```diff
@@ -0,0 +1,185 @@
+package app
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/mattermost/focalboard/server/model"
+	"github.com/mattermost/focalboard/server/services/auth"
+	"github.com/mattermost/focalboard/server/utils"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+var mockUser = &model.User{
+	ID:       utils.CreateGUID(),
+	Username: "testUsername",
+	Email:    "testEmail",
+	Password: auth.HashPassword("testPassword"),
+}
+
+func TestLogin(t *testing.T) {
+	th := SetupTestHelper(t)
+
+	testcases := []struct {
+		title    string
+		userName string
+		email    string
+		password string
+		mfa      string
+		isError  bool
+	}{
+		{"fail, missing login information", "", "", "", "", true},
+		{"fail, invalid username", "badUsername", "", "", "", true},
+		{"fail, invalid email", "", "badEmail", "", "", true},
+		{"fail, invalid password", "testUsername", "", "badPassword", "", true},
+		{"success, using username", "testUsername", "", "testPassword", "", false},
+		{"success, using email", "", "testEmail", "testPassword", "", false},
+	}
+
+	th.Store.EXPECT().GetUserByUsername(gomock.Eq("badUsername")).Return(nil, errors.New("Bad Username"))
```

Comment:
I'm not sure (I'll check), but I think if you are expecting one specific parameter you can use just `...GetUserByUsername("badUsername")...`